data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -0,0 +1,70 @@
+import { Case, CaseDocument } from '../model/case';
+import { NextFunction, Request, Response } from 'express';
+
+import { RevisionMetadata } from '../model/revision-metadata';
+
+const createNewMetadata = (curatorEmail: string) => {
+    return {
+        revisionNumber: 0,
+        creationMetadata: {
+            curator: curatorEmail,
+            date: Date.now(),
+        },
+    };
+};
+
+const createUpdateMetadata = (c: CaseDocument, curatorEmail: string) => {
+    return {
+        revisionNumber: ++c.revisionMetadata.revisionNumber,
+        creationMetadata: {
+            curator: c.revisionMetadata.creationMetadata.curator,
+            date: c.revisionMetadata.creationMetadata.date.getTime(),
+        },
+        updateMetadata: {
+            curator: curatorEmail,
+            date: Date.now(),
+        },
+    };
+};
+
+export const setRevisionMetadata = async (
+    request: Request,
+    response: Response,
+    next: NextFunction,
+) => {
+    const curatorEmail = request.body.curator.email;
+    const caseReference = request.body?.caseReference;
```

Comment:
why body? here and not above?

Reply:
just terrible inconsistency

it should always have a body at this point; otherwise openAPI would have rejected it